### PR TITLE
GFO-137/148: use a single Ticket component for /shop and /admin, user Tickets are displayed on the /shop endpoint

### PR DESCRIPTION
**Pull request by marcianszlovak**

### CODE REVIEW

File: frontend/src/components/TicketTypes/TicketList.jsx [hunk 1]
```diff
@@ -1,45 +1,42 @@
...
-            secret hacks to make .sass work with the ASCII portion of
-            grid-template-areas), would love to check out... Thanks in a
-          </div>
-          <Button buttonStyle="btn--primary--solid">BUY</Button>
-        </div>
+    <div className="ticket-list-container">
+      <div className="ticket-list">
+        <h1 className="main-title">Ticket Types</h1>
+        {ticketTypes.map(i => (
+          <AdminTicket key={i.id} {...i} />
```
Comment (adamgyulavari) [inline frontend/src/components/TicketTypes/TicketList.jsx:38]: @marcianszlovak this was not really the task :)
there should be Ticket component with configurable buttons
AdminTicket component should have these buttons, but here there should be a BUY button

**Comment by adamgyulavari** [comment]
@marcianszlovak also there's a conflict :)

File: frontend/src/components/ShopTicketList/ShopTicketList.jsx [hunk 2]
```diff
@@ -29,17 +29,22 @@ const ticketTypes = [
...
         <h1 className="main-title">Ticket Types</h1>
-        {ticketTypes.map(i => (
-          <AdminTicket key={i.id} {...i} />
+        {shopTicketTypes.map(shopTicket => (
+          <Ticket
+            key={shopTicket.id}
+            {...shopTicket}
+            canEdit={false}
+            buttonText="BUY"
+          />
```
Comment (adamgyulavari) [inline frontend/src/components/ShopTicketList/ShopTicketList.jsx:43]: @marcianszlovak maybe we should have the buttons as children elements:
```suggestion
          <Ticket
            key={shopTicket.id}
            {...shopTicket}
          >
          <Button buy />
          <Button edit />
          <...>
          </Ticket>
```

File: frontend/src/components/UserTicketList/UserTicketList.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <div className="ticket-list-container">
+      <div className="ticket-list">
+        <h1 className="main-title">My tickets</h1>
+        {userTicketTypes.map(userTicket => (
+          <Ticket
+            key={userTicket.id}
+            {...userTicket}
+            canEdit={false}
+            buttonText="SHOW"
+            title="Single ticket"
```
Comment (adamgyulavari) [inline frontend/src/components/UserTicketList/UserTicketList.jsx:33]: @marcianszlovak this should be in the data of `userTicket`

File: frontend/src/components/UserTicketList/UserTicketList.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import Ticket from '../Ticket/Ticket';
+
+const userTicketTypes = [
```
Comment (adamgyulavari) [inline frontend/src/components/UserTicketList/UserTicketList.jsx:4]: @marcianszlovak I guess this is the list of the tickets of users, so its not types
```suggestion
const userTickets = [
```

**Comment by marcianszlovak** [comment]
Not sure if these are the changes you suggested / recommended, please elaborate if not!

File: frontend/src/components/ShopTicketList/ShopTicketList.jsx (hunk 2 above)
Comment (adamgyulavari) [reply frontend/src/components/ShopTicketList/ShopTicketList.jsx]: @marcianszlovak don't put the button text and action into the mock data, just use children components in the `Ticket` component

**Comment by marcianszlovak** [comment]
If I pass the buttons as child components they simply don't render, not sure if this implementation is good, I'm probably missing something quite obvious.If I pass the buttons as child components they simply don't render, not sure if this implementation is good, I'm probably missing something quite obvious.

**Comment by adamgyulavari** [comment]
https://reactjs.org/docs/composition-vs-inheritance.html :)

**Comment by marcianszlovak** [comment]
OK, so I found out why it wasn't working before.  I tried what you linked before but thank you (seriously). All I had to do was use the spread operator in the ticket component to pass the props...

**Pull request merged**